demos/node-classification/ppnp/appnp-reddit.ipynb

```diff
@@ -0,0 +1,415 @@
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "152410 training nodes.\n",
+      "23699 validation nodes.\n",
+      "55334 testing nodes.\n"
+     ]
+    }
+   ],
+   "source": [
+    "def map_node_to_split(node):\n",
+    "    if node[\"test\"]:\n",
+    "        return \"test\"\n",
+    "    elif node[\"val\"]:\n",
+    "        return \"val\"\n",
+    "    else:\n",
+    "        return \"train\"\n",
+    "\n",
+    "\n",
+    "train_test_val_dict = dict(\n",
+    "    zip(list_node_ids, map(map_node_to_split, graph_data[\"nodes\"]))\n",
+    ")\n",
+    "\n",
+    "train_mask = [(train_test_val_dict[key] == \"train\") for key in list_node_ids]\n",
+    "val_mask = [(train_test_val_dict[key] == \"val\") for key in list_node_ids]\n",
+    "test_mask = [(train_test_val_dict[key] == \"test\") for key in list_node_ids]\n",
+    "\n",
+    "scaler = preprocessing.StandardScaler()\n",
+    "scaler.fit(node_data[train_mask].values)\n",
+    "node_data.iloc[:, :] = scaler.transform(node_data.values)\n",
+    "\n",
+    "train_data, train_targets = node_data[train_mask], targets[train_mask]\n",
+    "val_data, val_targets = node_data[val_mask], targets[val_mask]\n",
+    "test_data, test_targets = node_data[test_mask], targets[test_mask]\n",
+    "\n",
+    "print(\"{} training nodes.\".format(len(train_data)))\n",
+    "print(\"{} validation nodes.\".format(len(val_data)))\n",
+    "print(\"{} testing nodes.\".format(len(test_data)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating an adjacnecy matrix and StellarGraph generator\n",
+    "\n",
+    "We know create a `StellarGraph` object and use this to create the data generators."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Using GCN (local pooling) filters...\n"
+     ]
+    }
+   ],
+   "source": [
+    "G = sg.StellarGraph(\n",
+    "    nodes=node_data, edges=edge_df, source_column=\"source\", target_column=\"target\"\n",
+    ")\n",
+    "generator = FullBatchNodeGenerator(G, method=\"gcn\", sparse=True)\n",
+    "\n",
+    "test_gen = generator.flow(test_data.index, test_targets)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Training\n",
+    "\n",
+    "We now create a MLP and train on the node features."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": []
+    }
+   ],
+   "source": [
+    "in_layer = layers.Input(shape=(train_data.shape[-1]))\n",
+    "\n",
+    "layer = layers.Dense(512, activation=\"relu\", kernel_regularizer=\"l2\")(in_layer)\n",
+    "layer = layers.Dropout(0.5)(layer)\n",
+    "layer = layers.Dense(512, activation=\"relu\", kernel_regularizer=\"l2\")(in_layer)\n",
```

Comment:
Just checking, the input to this `Dense` layer should be `layer` output from the previous layer or the `in_layer` is the correct one?

Reply:
woops, nice catch! I'll fix that 👍